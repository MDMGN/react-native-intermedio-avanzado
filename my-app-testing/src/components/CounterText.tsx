import { View, Text } from "react-native";
import React from "react";
import useCounter from "../store/counter.store";

export default function CounterText() {
  const count = useCounter((state) => state.count);
  console.log("Render desde CounterText");
  return (
    <View>
      <Text testID="counter-text">{count}</Text>
    </View>
  );
}
