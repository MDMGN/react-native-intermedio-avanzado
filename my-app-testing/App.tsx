import { StyleSheet, View } from "react-native";
import CounterText from "./src/components/CounterText";
import Button from "./src/components/Button";

export default function App() {
  return (
    <View style={styles.container}>
      <CounterText />
      <Button />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
