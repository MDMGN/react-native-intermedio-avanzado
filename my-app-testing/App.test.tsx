import {
  act,
  fireEvent,
  render,
  renderHook,
  screen,
} from "@testing-library/react-native";
import useCounter from "./src/store/counter.store";
import CounterText from "./src/components/CounterText";
import App from "./App";

describe("Probando App.tsx", () => {
  it("Probando counter.store", () => {
    const { result } = renderHook(useCounter);
    expect(result.current.count).toBe(0);
  });

  it("Probando CounterText", async () => {
    render(<App />);
    const counterText = await screen.findByTestId("counter-text");
    const counterButton = await screen.findByTestId("counter-button");
    expect(counterText).toHaveTextContent("0");
    fireEvent.press(counterButton);
    expect(counterText).toHaveTextContent("1");
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
