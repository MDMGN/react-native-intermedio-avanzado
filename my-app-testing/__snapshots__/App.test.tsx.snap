// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando App.tsx Probando CounterText 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text
      testID="counter-text"
    >
      1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="counter-button"
  >
    <Text>
      +
    </Text>
  </View>
</View>
`;
